runtime/runtime/src/lib.rs

```diff
@@ -1512,15 +1508,15 @@ impl Runtime {
             &apply_state.config.congestion_control_config,
         );
 
+        receipt_sink.save_outgoing_buffer_metadatas(&mut processing_state.state_update);
+
```

Comment:
Sanity check
This here covers the new chunk case. The old chunk is an early returns a few lines above that. Do you not call it there because the metadata does not change for a missing chunk?